### ISSUE TITLE
Unknown instrument: Screenshot

### ISSUE THREAD

**Issue by d2boy** [HimbeersaftLP/ios-safari-remote-debug-kit#30]
Hi, I get connected to my IPAD, but get the following error 

Web Inspector encountered an internal error.
**These uncaught exceptions caused the problem:**
Unknown instrument: Screenshot (at : undefined: undefined)

![image](https://github.com/HimbeersaftLP/ios-safari-remote-debug-kit/assets/33113853/9edb942b-a8a0-4211-bb61-a7e9d98bb8a5)

**Comment by HimbeersaftLP** [comment]
Which iOS version and you using and which version did you run the generate script for?

**Comment by d2boy** [comment]
I have IOS 15, choose 17.4 while compiling script. Maybe that's it

**Comment by HimbeersaftLP** [comment]
Yeah, that is most likely the issue, the inspector is trying to use a feature that is not available on iOS 15, yet.
You'll always need to select a version that is <= the version you want to debug.
Please try that and let me know if it works.

**Comment by rcaliman-apple** [comment]
> I encountered the same error. I have iPhone 13 with iOS 17.6.1 which is not available on the list so I selected 17.4 for the build.

`Console.setConsoleClearAPIEnabled` [is available as of iOS 18.0 onwards](https://github.com/WebKit/WebKit/blob/33ad37aabfeee530c2a37e445fc32f5b9c524a8c/Source/WebInspectorUI/UserInterface/Controllers/ConsoleManager.js#L220-L221). That's why it fails for with the iOS 17.4 protocol snapshot.

@HimbeersaftLP This project doesn't seem to ever use the `InspectorBackendCommands.js` file that you get from building tip-of-tree WebKit. It seems to always prefer the latest protocol _snapshot_ from the repository which doesn't necessarily mean latest protocol version (which is what you get from building WebKit). This explains @prince-jagani's issue above.

We've been seeing a bunch of mysterious bugs about missing commands in Web Inspector filed on https://bugs.webkit.org. One [helpful commenter](https://bugs.webkit.org/show_bug.cgi?id=285693) pointed to the existence of this project.

The `InspectorFrontendHostStub.js` file in this repo is out of date and that causes other [types of bugs](https://bugs.webkit.org/show_bug.cgi?id=285111).

**Comment by HimbeersaftLP** [comment]
@rcaliman-apple Thanks a lot for the insight! I wasn't aware of the existence of that file in compiled WebKit and I'm sorry for my project causing this trouble in your issue tracker, I didn't think it would get popular 😅

I will update my tool to download the latest build from https://webkit.org/build-archives/#mac-sequoia-x86_64%20arm64 and extract the InspectorBackendCommands.js file from that.

I know that the stub is out of date as well (#35), I sadly haven't gotten around to updating it, yet, though.
